waspc/examples/todoApp/ext/Todo.js

```diff
@@ -98,12 +99,25 @@ const Tasks = (props) => {
 }
 
 const Task = (props) => {
+  const action = useAction(updateTaskIsDone, {
+    optimisticUpdates: [{
+      getQuery: () => [getTasks],
+      updateQuery: (updatedTask, oldTasks) => {
```

Comment:
Can (or should) we do something about incoherent UI after performing an optimistic update?

For example, let's assume that the server takes 3 seconds to update a task's status.

Checking several boxes, updating their status, and then clicking delete does nothing (because the server does not yet consider them complete).

Martin and I discussed this several weeks ago and the conclusion was that this behavior was a part of the application logic. The programmer can decide whether he wants to implement the "Delete" button:
- As a call to a "Delete all done" endpoint that queries the database and deletes the tasks _the database considers done_
- As a call to a "Delete by ID" endpoint that receives the IDs from the client and deletes the tasks _the client considers done_

This argument still makes sense to me, but I just wanted to document it.

Reply:
Interesting. The only other thing I can think of is if we exposed some way for them to register a callback so when all the in-progress calls for some cache key are done and the queries were invalidated, that they can run something. So they would disable the delete button perhaps, run the optimistic update, and then when they knew the server was all caught up they can run a callback to re-enable?